[Security Solution][Endpoint] Adjust usage info. (help output) for `memory-dump` and adds i18n'd Endpoint response codes

paul-tavares opened the pull request:
## Summary

The following changes were in support of response actions:

- Console Framework:
    - Added ability to override the "Usage" section content of the output of the `<command> --help`
    - Moved the `Examples` section of the command's `--help` output to be at the bottom of the output rather than right after `Usage` section
    - Deleted the Command Argument Definition schema property `selectorShowTextValue` - not necessary. Argument Render Components should be able to handle when the component is initialized with text data (as a result of either selecting a console's input history item or pasting the command into the console)
        - As a result of this, the `FilePicker` argument selector was adjusted to ensure proper initialization
- For the `memory-dump` response action (currently behind feature flag `responseActionsEndpointMemoryDump`):
    - Updated `memory-dump` command to include a custom `helpUsage` value in order to correctly show the usages set
    - Adds the `codes` that can be returned for a `memory-dump` response action to the list of `i18n`'d messages
    - Switched the order of mutually exclusive arguments so that `--process` is above `--kernel` (feedback from endpoint dev team)
    - Fixed a few warnings that are output by React due to state updates (often seen while running Jest tests or in the browser's console. Messages like: `Cannot update a component ('ComponentNameHere') while rendering a different component ('ComponentNameHere')`)


<br>

<img width="1153" height="596" alt="image" src="https://github.com/user-attachments/assets/ec43c65b-b5fa-4ac9-a741-b554fb83e4e3" />


<br><br>

### Checklist

- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

paul-tavares committed:
Rename `helpUsage` Command definition property to `helpExample`

paul-tavares committed:
Add support to `Console` for `helpUsage` override

paul-tavares committed:
Add `helpUsage` override to `memory-dump` command

paul-tavares committed:
add clarifying docs to types in `Console`

paul-tavares committed:
Add response codes for Memory Dump

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

paul-tavares committed:
Move the example section of Help output to the bottom

paul-tavares committed:
Remove `selectorShowTextValue` argument schema property (not needed)

paul-tavares committed:
Update `FileSelector` to handle initialization string values

paul-tavares committed:
Switch the order of --process and --kernel for memory dump

paul-tavares commented:
/ci

paul-tavares committed:
Fix console to avoid React state update warnings

'Warning: Cannot update a component (`CommandInput`) while rendering a different component (`ConsoleStateProvider`). To locate the bad setState() call inside `ConsoleStateProvider`'

paul-tavares committed:
Fix tests due to re-arrangement of help `Examples`

paul-tavares committed:
Fix additional updates of state mid-way through other updates

paul-tavares committed:
Merge branch 'main' into task/olm-14664-console-support-override-help-usage

paul-tavares committed:
Fix Mocks + tests

paul-tavares committed:
Merge remote-tracking branch 'origin/task/olm-14664-console-support-override-help-usage' into task/olm-14664-console-support-override-help-usage

paul-tavares committed:
Apply suggestions from code review

Co-authored-by: Leszek Kubik <39905449+intxgo@users.noreply.github.com>

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/362101)
* Commit: 7ed4e46a065feebd6f96459c2d11527dd8db89da
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-242605)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/194be51ac3aeabf7748a1af73318236dc09c657d) | [after](https://github.com/elastic/kibana/commit/7ed4e46a065feebd6f96459c2d11527dd8db89da) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 8379 | 8380 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/194be51ac3aeabf7748a1af73318236dc09c657d) | [after](https://github.com/elastic/kibana/commit/7ed4e46a065feebd6f96459c2d11527dd8db89da) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 11.1MB | 11.1MB | +2.4KB |



### History
* :broken_heart: [Build #361917](https://buildkite.com/elastic/kibana-pull-request/builds/361917) failed 14a1df6efd0c1fa9542fbdf7841c28082bd7fe5a
* :broken_heart: [Build #360663](https://buildkite.com/elastic/kibana-pull-request/builds/360663) failed f196f46aa57e68000fa7e4f05e6d0cb3fb665006

cc @paul-tavares

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/362101","number":362101,"commit":"7ed4e46a065feebd6f96459c2d11527dd8db89da"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361917","number":361917,"commit":"14a1df6efd0c1fa9542fbdf7841c28082bd7fe5a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/360663","number":360663,"commit":"f196f46aa57e68000fa7e4f05e6d0cb3fb665006"}],"number":362101}
buildkite-pr-comment-->

paul-tavares committed:
Merge branch 'main' into task/olm-14664-console-support-override-help-usage

ashokaditya commented on x-pack/solutions/security/plugins/security_solution/public/management/components/endpoint_responder/lib/endpoint_action_response_codes.ts:
```diff
@@ -332,6 +332,67 @@ const CODES = Object.freeze({
+  // DEV: process with pid or entity_id not foud
+  'ra_memory-dump_error_not-found': i18n.translate(
+    'xpack.securitySolution.endpointActionResponseCodes.memoryDump.notFound',
+    { defaultMessage: 'The process to dump was not found' }
```
```suggestion
    { defaultMessage: 'The process for memory dump was not found' }
```

ashokaditya reviewed (approved):
Thanks for the changes. I tested it and works as expected. 

Not related to changes in this PR:
I noticed that the help shortcut button for `memory-dump --process` does not input a default parameter (--pid or --entityId) with it, similar to how we do or kill and suspend process actions. Maybe we should make it consistent and enter the most likely param (--pid) when we use the shortcut button (`memory-dump --process --pid`).

paul-tavares committed:
Update x-pack/solutions/security/plugins/security_solution/public/management/components/endpoint_responder/lib/endpoint_action_response_codes.ts

Co-authored-by: Ash <1849116+ashokaditya@users.noreply.github.com>

paul-tavares committed:
Merge branch 'main' into task/olm-14664-console-support-override-help-usage
